backend/bot/index.js

```diff
@@ -36,12 +62,48 @@ const postResultToSlack = (match) => {
     prefix = 'HOLY SHIT! ';
   }
 
-  const suffix = isComedyDuo?  ':marioluigi:' : '';
+  const suffix = isComedyDuo ? ':marioluigi:' : '';
 
   const messageText = `${prefix}${winningPlayers.join(', ')} just beat ${losingPlayers.join(', ')}. Rating change: ${ratingChange}${suffix}`
-  bot.postMessage(channel.id, messageText, {as_user: true});
+  await bot.postMessage(channel.id, messageText, { as_user: true });
+}
+
+const postRankingChangeMessage = async (oldRankings, newRankings) => {
+  let shouldUpdatePurpose = false
+  const rankingChanges = []
+  oldRankings.forEach((oldPlayer, index) => {
+    const newPlayer = newRankings[index]
+    if (newPlayer.id !== oldPlayer.id) {
+      if (index <= 4) {
+        shouldUpdatePurpose = true
+      }
+      rankingChanges.push({
+        name: oldPlayer.name,
+        oldRanking: index +1 ,
+        newRanking: newRankings.findIndex(p => p.id === oldPlayer.id) + 1
+      })
+    }
+  })
+  const messageText = rankingChanges
+    .map((c) => (
+      `${c.name} ${c.oldRanking}. ⟶ ${c.newRanking}.`)
+    )
+    .join('\n')
+  await bot.postMessage(channel.id, messageText, { as_user: true });
+
+  return shouldUpdatePurpose
 }
 
+const updatePurpose = async (rankings) => {
+  const purpose = ['TOP PLAYERS:']
+  for (let i = 0; i < 5; i++) {
```

Comment:
this `< 5` is tightly coupled with "<= 4" in `postRankingChangeMessage`, yet it's in a different function

Reply:
i'd make updatePurpose dumb printing out the whole array provided